YFUtechne.sol

```diff
@@ -2,56 +2,65 @@
 pragma solidity ^0.8.4;
 
 import "@openzeppelin/contracts/token/ERC721/ERC721.sol";
-import "@openzeppelin/contracts/access/AccessControl.sol";
+import "@openzeppelin/contracts/access/Ownable.sol";
 import "@openzeppelin/contracts/utils/Counters.sol";
+import "@openzeppelin/contracts/interfaces/IERC20.sol";
 
-contract YFUtechne is ERC721, AccessControl {
+contract YFUtechne is ERC721, Ownable {
     using Counters for Counters.Counter;
 
     Counters.Counter private _tokenIdCounter;
 
-    uint256 public MAX_SUPPLY = 10;
-    uint256 public PRICE = 1 ether;
+    uint256 immutable internal MAX_SUPPLY = 10;
+    uint256 immutable internal PRICE = 1 ether;
     address payable public depositAddress;
     bool public transfers_frozen = true;
     string public ipfsBaseURI;
 
     constructor(address payable adminAddress, string memory ipfsURI) ERC721("YFU Techne", "YFU_1") {
         ipfsBaseURI = ipfsURI;
         depositAddress = adminAddress;
-        _grantRole(DEFAULT_ADMIN_ROLE, adminAddress);
     }
 
-    function set_ipfs_base_uri(string memory ipfsURI) public onlyRole(DEFAULT_ADMIN_ROLE) {
+    function set_ipfs_base_uri(string memory ipfsURI) external onlyOwner {
         ipfsBaseURI = ipfsURI;
     }
 
-    function set_deposit_address(address payable to) public onlyRole(DEFAULT_ADMIN_ROLE) {
+    function set_deposit_address(address payable to) external onlyOwner {
         depositAddress = to;
     }
 
-    function unfreeze_transfers() public onlyRole(DEFAULT_ADMIN_ROLE) {
+    function unfreeze_transfers() external onlyOwner {
         transfers_frozen = false;
     }
 
     function _baseURI() internal view override returns (string memory) {
         return ipfsBaseURI;
     }
 
-    function safeMint(address to) public payable {
+    function safeMint(address to) external payable {
         require(_tokenIdCounter.current() < MAX_SUPPLY, "Maximum token supply reached");
         require(msg.value == PRICE, "Invalid amount");
 
-        depositAddress.transfer(PRICE);
-
         uint256 tokenId = _tokenIdCounter.current();
         _tokenIdCounter.increment();
         _safeMint(to, tokenId);
     }
 
+    function withdraw() external onlyOwner {
+        uint256 balance = address(this).balance;
+        depositAddress.call{value: balance}("");
```

Comment:
use the return value of `.call` to determine success. The report gives a reference for this.